microsoft/botframework-solutions
# Lamil/dispatch by locale

## Description

Enabling multiple language configurations in a single bot instance.

Notable changes/improvements include:
- Powershell script for deploying one bot with multiple language config files
- Powershell script to check for duplicate utterances in .lu files and updating deployment scripts
- Powershell script to update all the published models based on .lu files (saves previous version as "backup" in case of issues)
- Bot logic to support multiple language configs in VA and skills

## Related Issue




closes #275 

## Testing Steps



### Test deployment
- Run deploy_bot.ps1 script in DeploymentScripts folder (By default, this deploys all languages. You can change this by adding the locales parameter: `-locales "en-us,zh-cn"`)
- Update appSettings with your locale .bot files
```
 "defaultLocale": "en-us",
  "languageModels": {
    "en": {
      "botFilePath": ".\\LocaleConfigurations\\YOUR_LOCALE_BOT_PATH.bot",
      "botFileSecret": ""
    }
  },
```
- Set up an auth connection for your bot
- Run the project, and specify different locales in the emulator settings page to test different languages (NOTE: responses are not currently available for all languages, but LUIS, intro, and help cards should be localized)

### Test updating
- Make a change to one of the .lu files
- run update_published_models.ps1 in DeploymentScripts
- Verify your change is active in the LUIS portal and the "backup" version has been created

## Checklist

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
~~- [ ] A duplicate issue is filed to track future  work.~~

If you have updated responses or `.lu` files:
